python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -24,152 +26,98 @@ class VxlanInterfaceMixin(Protocol):
                 continue
 
             for l2vlan in tenant.l2vlans:
-                if vlan := self._get_vxlan_interface_config_for_vlan(l2vlan, tenant):
-                    # Duplicate check is not done on the actual list of vlans, but instead on our local "vnis" list.
-                    # This is necessary to find duplicate VNIs across multiple object types.
-                    append_if_not_duplicate(
-                        list_of_dicts=vnis,
-                        primary_key="vni",
-                        new_dict=vlan,
-                        context="VXLAN VNIs for L2VLANs",
-                        context_keys=["id", "name", "vni"],
-                    )
-                    # Here we append to the actual list of VRFs, so duplication check is on the VLAN ID here.
-                    append_if_not_duplicate(
-                        list_of_dicts=vlans,
-                        primary_key="id",
-                        new_dict=vlan,
-                        context="VXLAN VNIs for L2VLANs",
-                        context_keys=["id", "vni"],
-                    )
+                if l2vlan.vxlan:
+                    vlan = self._get_vxlan_interface_config_for_vlan(l2vlan, tenant)
+                    self._check_for_duplicate(vlan, vnis)
+                    vxlan.vlans.append(vlan)
 
         if self.shared_utils.is_wan_server:
             # loop through wan_vrfs and add VRF VNI if not present
             for vrf in self._filtered_wan_vrfs:
-                # Duplicate check is not done on the actual list of vlans, but instead on our local "vnis" list.
-                # This is necessary to find duplicate VNIs across multiple object types.
-                vrf_data = {"name": vrf.name, "vni": vrf.wan_vni}
-                append_if_not_duplicate(
-                    list_of_dicts=vnis,
-                    primary_key="vni",
-                    new_dict=vrf_data,
-                    context="VXLAN VNIs for VRFs",
-                    context_keys=["id", "name", "vni"],
-                )
-                # Here we append to the actual list of VRFs, so duplication check is on the VRF here.
-                append_if_not_duplicate(
-                    list_of_dicts=vrfs,
-                    primary_key="name",
-                    new_dict=vrf_data,
-                    context="VXLAN VNIs for VRFs",
-                    context_keys=["name", "vni"],
-                )
-
-        if vlans:
-            vxlan["vlans"] = vlans
-
-        if vrfs:
-            vxlan["vrfs"] = vrfs
-
-        return {
-            "vxlan1": {
-                "description": f"{self.shared_utils.hostname}_VTEP",
-                "vxlan": vxlan,
-            },
-        }
-
-    def _get_vxlan_interface_config_for_vrf(
+                new_vrf = EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan.VrfsItem(name=vrf.name, vni=vrf.wan_vni)
+                self._check_for_duplicate(new_vrf, vnis)
+                vxlan.vrfs.append(new_vrf)
```

Comment:
We could just append to the final list everywhere and then have a single function that parses all the lists once and check for duplicate VNIs in the end?